server/bleep/src/webserver/studio.rs

```diff
@@ -389,3 +396,94 @@ async fn generate_llm_context(app: Application, context: Vec<ContextFile>) -> Re
 
     Ok(s)
 }
+
+#[derive(serde::Deserialize)]
+pub struct Import {
+    pub thread_id: Uuid,
+}
+
+/// Returns a new studio UUID.
+pub async fn import(
+    app: Extension<Application>,
+    user: Extension<User>,
+    Query(params): Query<Import>,
+) -> webserver::Result<String> {
+    let user_id = user
+        .login()
+        .ok_or_else(|| super::Error::user("didn't have user ID"))?
+        .to_string();
+
+    let thread_id = params.thread_id.to_string();
+
+    let conversation = sqlx::query! {
+        "SELECT title, repo_ref, exchanges
+         FROM conversations
+         WHERE user_id = ? AND thread_id = ?",
+        user_id,
+        thread_id,
+    }
+    .fetch_optional(&*app.sql)
+    .await
+    .map_err(Error::internal)?
+    .ok_or_else(|| Error::new(ErrorKind::NotFound, "conversation not found"))?;
+
+    let repo_ref = conversation.repo_ref;
+    let exchanges = serde_json::from_str::<Vec<Exchange>>(&conversation.exchanges)
+        .context("couldn't deserialize exchange list")?;
+
+    let context = exchanges
+        .iter()
+        .flat_map(|e| {
+            let mut chunk_map = HashMap::new();
+
+            for c in &e.code_chunks {
+                chunk_map
+                    .entry(&c.path)
+                    .or_insert_with(Vec::new)
+                    .push(c.start_line..c.end_line + 1);
+            }
+
+            chunk_map.into_iter().map(|(path, ranges)| ContextFile {
+                path: path.clone(),
+                hidden: false,
+                repo: repo_ref.parse().unwrap(),
+                branch: e.query.branch().next().map(Cow::into_owned),
+                ranges,
+            })
+        })
+        .collect::<Vec<_>>();
+
+    let messages = exchanges
+        .iter()
+        .filter_map(|e| {
+            let query = e.query.target()?;
```

Comment:
```suggestion
            let query = e.query()?;
```

`Exchange::query` works similarly.